hoomd/hpmc/compute.py

```diff
@@ -325,18 +430,37 @@ def betaP(self):  # noqa: N802 - allow function name
             In MPI parallel execution, `betaP` is available on rank 0 only.
             `betaP` is `None` on ranks >= 1.
         """
-        if not numpy.isnan(self.sdf).all():
+        if not (numpy.isnan(self.sdf_compression).all()
+                or numpy.isnan(self.sdf_expansion).all()):
+            compression_contribution = 0
+            expansion_contribution = 0
+            box = self._simulation.state.box
+            N = self._simulation.state.N_particles
+            rho = N / box.volume
+
+            # compressive contribution
             # get the values to fit
-            sdf_fit = numpy.array(self.sdf)
-            # construct the x coordinates
-            x_fit = numpy.arange(0, len(sdf_fit), 1) * self.dx
+            sdf_fit_compression = numpy.array(self.sdf_compression)
+            x_fit = numpy.arange(0, len(sdf_fit_compression), 1) * self.dx
             x_fit += self.dx / 2
+            # construct the x coordinates
+            # perform the fit and extrapolation
+            p = numpy.polyfit(x_fit, sdf_fit_compression, 5)
+            p0_compression = numpy.polyval(p, 0.0)
+            compression_contribution = rho * p0_compression / (2
+                                                               * box.dimensions)
+
+            # expansive contribution
+            # reverse the x-coordinates
+            x_fit = -x_fit[::-1]
```

Comment:
This assumes that `len(sdf_compression) == len(sdf_expansion)`. Perhaps add an assert here to be certain.